analyzers/src/SonarAnalyzer.CSharp/Rules/MemberShadowsOuterStaticMember.cs

```diff
@@ -102,135 +88,60 @@ protected override void Initialize(SonarAnalysisContext context)
                 return;
             }
 
-            foreach (var reference in namedType.DeclaringSyntaxReferences)
+            var namedTypeName = namedType.TypeKind switch { TypeKind.Class => "class", TypeKind.Delegate => "delegate" };
```

Comment:
From semantic of the algorithm, there's a question `else what?` The extraction was a good catch, but should follow the original idea:
```suggestion
            var namedTypeName = namedType.TypeKind == TypeKind.Delegate ? "delegate" : "class";
```

Reply:
`namedTypeName` sounds too wordy. Can it be something like `kindName` instead?